src/main/java/com/chrosciu/bootcamp/tasks/github/GithubClient.java

```diff
@@ -4,28 +4,29 @@
 import com.chrosciu.bootcamp.tasks.github.dto.Repository;
 import lombok.RequiredArgsConstructor;
 import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Schedulers;
 
 @RequiredArgsConstructor
 public class GithubClient {
     private final GithubApi githubApi;
 
     public Flux<Repository> getUserRepositories(String username) {
-        //TODO: Implement
-        return null;
+        return githubApi.getUserRepositories(username).flatMapMany(Flux::fromIterable).onErrorReturn(new Repository());
     }
 
     public Flux<Branch> getUserRepositoryBranches(String username, String repo) {
-        //TODO: Implement
-        return null;
+        return githubApi.getUserRepositoryBranches(username, repo).flatMapMany(Flux::fromIterable);
     }
 
     public Flux<Repository> getUsersRepositories(Flux<String> usernames) {
-        //TODO: Implement
-        return null;
+        return usernames.flatMap(this::getUserRepositories);
     }
 
     public Flux<String> getAllUserBranchesNames(String username) {
-        //TODO: Implement
-        return null;
+        return getUserRepositories(username)
```

Comment:
No niestety nie tędy droga (ale miło że znalazłeś to w dokumentacji:)) Ten kod dalej nie wykonuje się  równolegle - natomiast już niewiele Cię od tego dzieliło. Paradoksalnie problemem jest użycie operatora parallel() - ze względu na to iż powoduje on stworzenie ParallelFlux, z którego nie ma potem żadnego pożytku (po operatorze parallel nie ma tak naprawdę już żadnej operacji, którą dałoby się zrównoleglić - a co gorsza taki ParallelFlux trzeba potem po cichu scalić żeby dało się go użyć jako wynik wykonania lambdy we flatMap co zżera cały zysk z równoleglenia). Drugą sprawą jest użycie Schedulers.parallel() - przy operacjach blokujących może to spowodować kłopoty ze względu na sztywną ilość wątków takiego schedulera - tu lepszy jest Schedulers.elastic().